odoo/orm/models.py

```diff
@@ -5629,6 +5629,14 @@ def ensure_one(self) -> Self:
         except ValueError:
             raise ValueError("Expected singleton: %s" % self)
 
+    @api.private
+    def ensure_cursor_read_write(self) -> Self:
```

Comment:
It is quite low level and trivial. I think we don't need this method.
(also it does not return Self)

```py
if self.env.cr.readonly:
    raise ReadOnlyError()
```